### PR TITLE
Improve //search algorithm

### PR DESCRIPTION
**Pull request by Kubo2**
Just code-level changes.

### CODE REVIEW

File: welcomebot.py [hunk 1]
```diff
@@ -187,24 +187,31 @@ def perform_image_search():
...
             def perform_google_search():
                 search_term = "+".join(message.content.split()[1:])
-                search_results = google_search.google_search(search_term)
-                print search_results
-                if search_results is False:
-                    print "No Results"
+                try:
+                    results, moreResultsUrl = google_search.google_search(search_term)
+                    print results
+                except TypeError:
```
Comment (thomas-daniels) [inline welcomebot.py:197]: If you cannot extract enough values from a return value, Python throws a ValueError, not a TypeError.

**Comment by thomas-daniels** [comment]
See my line note. Also, the users.txt change is going to give painful merge conflicts, and it isn't really necessary to have it changed in this pull request. Do you mind undoing the changes there? Thanks!

File: welcomebot.py (hunk 1 above)
Comment (Kubo2) [inline welcomebot.py:197]: In the time of writing this piece of code, I was not sure about it, so I have tried the following:

``` bash
kubo2@kubo2:~/WelcomeBot$ python -c 'a, b = False'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: 'bool' object is not iterable
kubo2@kubo2:~/WelcomeBot$ python --version
Python 2.7.6
```

The same it behaves when feeding up `a` and `b` from a function return value. You have to realize, that when you're doing `return False`, you're **not returning a tuple** (which is iterable) with not enough values, you're just simply returning a non-iterable `bool` object of value `False`.

Comment (thomas-daniels) [inline welcomebot.py:197]: Hmm, you're right indeed. Thanks for correcting me!

**Comment by Kubo2** [comment]
Hey @ProgramFOX!

Thanks for your feedback. I didn't notice that I have been commiting the
users.txt file (I have changed it and then ran `cp -R WelcomeBot/ Kubobot/`
to rather run Kubobot's own copy of WelcomeBot, so I forgot to `git reset
HEAD` it), eitherway, it is a mistake and will fix it. (I also don't
mention that it was about 2am in Slovakia when I was commiting it. :-)

So will fix and force-push in minute or two.

Best regards,
Kubo2

2015-07-30 9:20 GMT+02:00 ProgramFOX notifications@github.com:

**Comment by Kubo2** [comment]
Hey @ProgramFOX, amended and pushed `master`'s version of `users.txt`.

**Comment by thomas-daniels** [comment]
Great, thanks!

**Pull request merged**